src/actions/details.js

```diff
@@ -0,0 +1,36 @@
+const DETAILS_FETCH_REQ = 'DETAILS_FETCH_REQ'
+const DETAILS_FETCH_START = 'DETAILS_FETCH_START'
+const DETAILS_FETCH_SUCCESS = 'DETAILS_FETCH_SUCCESS'
+const DETAILS_FETCH_FAIL = 'DETAILS_FETCH_FAIL'
+
+
+const fetchDetails = data => ({
+  type: DETAILS_FETCH_REQ,
+  payload: data
```

Comment:
As a note, it is a good practice to wrap the payload inside an object (this convention was first created by the `Flux` team, and it's recommend to use also in `Redux` for consistency).

```js
payload: { gif },
```